src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1561,6 +1574,48 @@ protected Object job(final SQLiteConnection sqliteConnection) throws CatalogExce
   }
 
   /**
+   * Update the partition function of an ingested dataset.
+   * 
+   * @param key the relation key.
+   * @param howPartitioned how the dataset was partitioned.
+   * @throws CatalogException if there is an error in the catalog.
+   */
+  public void updateHowPartitioned(final RelationKey key, final HowPartitioned howPartitioned) throws CatalogException {
+    if (isClosed) {
+      throw new CatalogException("Catalog is closed.");
+    }
+
+    try {
+      queue.execute(new SQLiteJob<DatasetStatus>() {
+        @Override
+        protected DatasetStatus job(final SQLiteConnection sqliteConnection) throws CatalogException, SQLiteException {
+          try {
+            SQLiteStatement statement =
+                sqliteConnection
+                    .prepare("UPDATE stored_relations set how_partitioned=? WHERE user_name=? AND program_name=? AND relation_name=?");
+            try {
+              statement.bind(1, MyriaJsonMapperProvider.getMapper().writeValueAsString(howPartitioned));
+            } catch (JsonProcessingException e) {
+              throw new CatalogException(e);
+            }
+            statement.bind(2, key.getUserName());
+            statement.bind(3, key.getProgramName());
+            statement.bind(4, key.getRelationName());
+            statement.stepThrough();
+            statement.dispose();
+            statement = null;
+          } catch (final SQLiteException e) {
+            throw new CatalogException(e);
+          }
+          return null;
+        }
+      }).get();
+    } catch (InterruptedException | ExecutionException e) {
```

Comment:
I think this piece of code is different from the link that you gave. Here, `queue` is a `SQLiteQueue` object which executes jobs in a separated thread. This `InterruptedException` indicates that the SQLite thread was interrupted. The thread running code in `MasterCatalog` shouldn't be in the `interrupted` state. Here, the semantics is: something is wrong when updating the catalog, so just throw a `CatalogException` and return, instead of: the thread executing `updateHowPartitioned()` should stop and do something else -- but up to the caller to decide.